extensions/wikia/WallNotifications/WallNotificationEntity.class.php

```diff
@@ -78,174 +127,202 @@ public function getUniqueId() {
 		}
 	}
 
-	public function getId() {
-		return $this->id;
+	/**
```

Comment:
This is the big fix.  Call ```loadDataFromRevId``` on the wiki given by ```$wikiId``` when ```$wikiId``` is not the current wiki.